recipes/tcb-span/all/conanfile.py

```diff
@@ -0,0 +1,32 @@
+from conans import ConanFile, tools
+import os
+import glob
+
+
+class TcbSpanConan(ConanFile):
+    name = "tcb-span"
+    description = "Implementation of C++20's std::span for older compilers"
+    topics = ("conan", "span", "header-only", "tcb-span")
+    homepage = "https://github.com/tcbrindle/span"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "BSL-1.0"
+    settings = "os", "compiler", "build_type", "arch"
+
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
```

Comment:
You can simplify all of this, just use:

```suggestion
        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

---

If you follow the previous steps locally, you would find that the source code is extracted (usually) inside a folder.... in the recipe where you copied these lines from, that folder was something like the reference name followed by `-`. Then people usually rename the source folder to something known (and standard) for every recipe.